shortcodes/epfl-allowed-iframe/epfl-allowed-iframe.php

```diff
@@ -3,35 +3,76 @@
 namespace EPFL\Plugins\Shortcodes\EPFLAllowedIframe;
 use \EPFL\Plugins\Gutenberg\Lib\Utils;
 
-function epfl_allowed_iframe_process_shortcode( $atts, $content = null ) {
-    $atts = shortcode_atts( array(
-            'url' => ''
-    ), $atts );
 
-    if (strpos(file_get_contents("./wp-content/plugins/wp-gutenberg-epfl/shortcodes/epfl-allowed-iframe/allowed_url.txt"),$atts['url']) !== false) {
-    	$url = sanitize_text_field($atts['url']);
+// Logic to validate the sortcode URL attribute with the whitelist
+function allowed_url_checker( $url, $allowed_url ) {
+    if ( false === filter_var( $url, FILTER_VALIDATE_URL ) ) {
+        return false;
+    }
+    if ( false === filter_var( $allowed_url, FILTER_VALIDATE_URL ) ) {
+        return false;
     }
-    else
-    {
-       if ( $screen->parent_base != 'edit' ) {
-          return Utils::render_user_msg("iframe url not allowed");
-       }	
+    $url = parse_url( $url );
+    // HTTPS only
+    if ( ( isset( $url['scheme'] ) && 'https' !== $url['scheme'] ) || empty( $url['scheme'] ) ) {
+        return false;
     }
-?>
 
-<div class="my-3 container">
-    <div class="embed-responsive embed-responsive-16by9">
-	<iframe src="<?php echo esc_url($url); ?>" webkitallowfullscreen mozallowfullscreen allowfullscreen allow="autoplay; encrypted-media" frameborder="0" class="embed-responsive-item"></iframe>
-    </div>
-</div>
+    $allowed_url = parse_url( $allowed_url );
+    // host has to be the same
+    if ( $allowed_url['host'] !== $url['host'] ) {
+        return false;
+    }
 
-<?php
+    // this prevent to chech the path and the query string
+    // use: wp option add epfl_allowed_iframe_wildcard 1
+    if ( get_option('epfl_allowed_iframe_wildcard') != '1' ) {
```

Comment:
This should be a setting per line. The wwp operators may wish to allow wildcards for a given hostname, and whitelist specific URLs otherwise.

Please design and implement some kind of syntax for wildcard matching. For instance, lines in the config file that start end end with a pipe sign (|) could mean that the line is a regexp, which should be validated using preg_matc() rather than equality or host equality.

Reply:
Let's talk about it — while I completely agree with your proposal, I still have some design issues :
  - why the whitelist is common to all the websites ? We may want to grant some authorization for a website only.
  - would it be possible to manage the list outside of the code ? (wp-veritas?)

As it, this PR provide the same set of features as before.

Reply:
Additional thought: allow wildcards by default on the epfl.ch site, be more precise on external websites.